linera-views/src/collection_view.rs

```diff
@@ -513,6 +531,51 @@ where
     }
 }
 
+impl<C, W> ByteCollectionView<C, W>
+where
+    C: Context + Send,
+    ViewError: From<C::Error>,
+    W: HashableView<C> + Sync,
+{
+    /// Computes the hash of the view
+    async fn compute_hash_internal(&self) -> Result<<sha3::Sha3_256 as Hasher>::Output, ViewError> {
+        let mut hasher = sha3::Sha3_256::default();
+        let keys = self.keys().await?;
+        hasher.update_with_bcs_bytes(&keys.len())?;
+        let updates = self.updates.write().await;
+        for key in keys {
+            hasher.update_with_bytes(&key)?;
+            let hash = match updates.get(&key) {
+                Some(entry) => {
+                    let Update::Set(view) = entry else {
+                        unreachable!();
+                    };
+                    view.hash().await?
+                }
+                None => {
+                    let key = self.context.base_tag_index(KeyTag::Subview as u8, &key);
+                    let context = self.context.clone_with_base_key(key);
+                    let view = W::load(context).await?;
+                    view.hash().await?
+                }
+            };
+            hasher.write_all(hash.as_ref())?;
+        }
+        Ok(hasher.finalize())
+    }
+
+    async fn compute_hash(&self) -> Result<<sha3::Sha3_256 as Hasher>::Output, ViewError> {
+        #[cfg(feature = "metrics")]
+        return run_with_execution_time_metric(
```

Comment:
Shouldn't the tag ` #[cfg(feature = "metrics")]` be a the definition site of `run_with_execution_metric` (possibly marked as `inline`)?

Reply:
Mmm, but it would have be a macro, so that the last argument is not always needed. so I don't know